docs/source/examples/dcgan.rst

```diff
@@ -0,0 +1,358 @@
+DCGAN: Generate the images with Deep Convolutional GAN
+******************************************************
+
+.. currentmodule:: chainer
+
+0. Introduction
+================
+
+In this tutorial, we generate images with **generative adversarial network (GAN)**.
+It is a kind of generative model with deep neural network, and often applied to
+the image generation. The GAN technique is also applied to 
+`PaintsChainer <https://paintschainer.preferred.tech/index_en.html>`_,
+a famous automatic coloring service.
+
+.. image:: ../../image/dcgan/generated-images.gif
+
+In the tutorial, you will learn the following things:
+
+1. Generative Adversarial Networks (GAN)
+2. Implementation of DCGAN in Chainer
+
+1. Generarive Adversarial Networks (GAN)
+=========================================
+
+1.1 What is the GAN?
+---------------------
+
+As explained in the GAN tutorial in NIPS 2016 [1], the generative models can be classified into the categories as shown in the following figure:
+
+.. figure:: ../../image/dcgan/class-generative-model.png
+   :scale: 100%
+
+   cited from [1]
+
+Besides GAN, other famous generative models are Fully visible belief networks (FVBNs) and Variational autoencoder (VAE).
+Unlike FVBNs and VAE, GAN does not explicitly models the probability distribution :math:`p({\bf s})`
+that generates the training data. Instead, we model a generator :math:`G: {\bf z} \mapsto {\bf s}`.
+The generator :math:`G` samples :math:`{\bf s} \sim p({\bf s})` from the latent variable :math:`\bf z`.
+Apart from the generator :math:`G`, we create a discriminator :math:`D({\bf x})`
+which identified the samples from the generator :math:`G` and the true samples from training data.
+While training the discriminator :math:`D`, the generator :math:`G` is also trained so that
+the generated samples cannot be identified by the discriminator.
+The advantages of GAN are low sampling cost and state-of-the-art in image generation.
+The disadvantage is that we cannot calculate the probability distribution
+:math:`p_{\mathrm {model}}({\bf s})` because we do not model any probability distribution,
+and we cannot infer the latent variable :math:`\bf z` from a sample.
+
+1.2 How the GAN works?
+-----------------------
+
+As explained above, GAN uses the two models, the generator and the discriminator. In other words,
+we set up two neural networks for GAN.
+
+When training the networks, we should match the distribution of the samples
+:math:`{\bf s} \sim p({\bf s})` generated from the true distribution with the distribution of the samples
+:math:`{\bf s} = G ({\bf z})` generated from the generator.
+
+.. image:: ../../image/dcgan/gan-overview.png
+
+The generator :math:`G` learns the target distribution on the idea of
+**Nash equilibrium** [2] of game theory.
+In detail, while training the discriminator :math:`D`,
+the generator :math:`G` is also trained so that the discriminator :math:`D`
+cannot identify the generated samples.
```

Comment:
I rephrased as follows:
> In detail, while training the discriminator :math:`D`,
> the generator :math:`G` is also trained, so that
> the discriminator :math:`D` making a mistake.